Source/WebCore/platform/network/curl/CurlRequest.h

```diff
@@ -159,8 +152,8 @@ class CurlRequest : public ThreadSafeRefCounted<CurlRequest>, public CurlRequest
     bool m_didReceiveResponse { false };
     bool m_didNotifyResponse { false };
     bool m_didReturnFromNotify { false };
-    Action m_actionAfterInvoke { Action::None };
-    CURLcode m_finishedResultCode { CURLE_OK };
+    bool m_mustInvokeCancelTransfer { false };
+    Function<void()> m_responseCompletionHandler;
```

Comment:
How about `CompletionHandler` instead of `Function`?

Reply:
It seems that `CompletionHandler` must always call `operator()`. Otherwise an ASSERT is raised. On the other hand, completeDidReceiveResponse() is not always called after calling invokeDidReceiveResponse(). (For example, when it is canceled by CORS etc.) Therefore, `CompletionHandler` cannot be used in this case.

https://github.com/WebKit/WebKit/blob/dd1484feba3d6e44d5ac12d9a6162d47582781d4/Source/WTF/wtf/CompletionHandler.h#L65